src/Query/Builder.php

```diff
@@ -993,17 +994,15 @@ protected function compileWheres(): array
             }
 
             // Convert column name to string to use as array key
-            if (isset($where['column']) && $where['column'] instanceof Stringable) {
+            if (isset($where['column'])) {
                 $where['column'] = (string) $where['column'];
             }
 
             // Convert id's.
             if (isset($where['column']) && ($where['column'] === '_id' || str_ends_with($where['column'], '._id'))) {
                 if (isset($where['values'])) {
                     // Multiple values.
-                    foreach ($where['values'] as &$value) {
```

Comment:
This passing of the value by reference confused me.

Before the new loop, `$result = ['$in' => [ObjectId, ObjectId, '']]`.
<img width="906" alt="Screenshot 2023-10-17 at 23 38 41" src="https://github.com/mongodb/laravel-mongodb/assets/400034/a124c7d2-1e6b-41b2-b5ca-52b624960a15">

When the loop starts, `$result = ['$in' => [ObjectId, ObjectId, &$result]]`. With recursion.
<img width="1009" alt="Screenshot 2023-10-17 at 23 39 10" src="https://github.com/mongodb/laravel-mongodb/assets/400034/48a301f3-b7fb-4787-bc22-a575e45275fb">

Reply:
Yeah, foreach with references is always dodgy, I appreciate the refactoring 👍